mantis/solver/src/types.rs

```diff
@@ -0,0 +1,203 @@
+//! Basic types with simple checks and domain, no heavy math or solving.
+use crate::prelude::*;
+
+pub type Amount = Decimal;
+
+#[derive(Debug, Clone, Copy, Ord, Eq, PartialEq, PartialOrd, Default)]
+pub struct BuyToken(pub Amount);
+#[derive(Debug, Clone, Copy, Ord, Eq, PartialEq, PartialOrd, Default)]
+pub struct SellToken(pub Amount);
+
+#[derive(Debug, Clone, Copy)]
+pub struct Price(pub Amount);
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum OrderType {
+    BUY,
+    SELL,
```

Comment:
Since it’s being ported to Rust, let’s use Rust naming style as well:

```suggestion
    Buy,
    Sell,
```

Reply:
will retain python notation for now